components/raftstore/src/store/fsm/peer.rs

```diff
@@ -6258,6 +6341,42 @@ where
         self.fsm.peer.leader_lease_mut().expire_remote_lease();
     }
 
+    fn on_ready_batch_switch_witness(&mut self, sw: SwitchWitness) {
+        {
+            let mut meta = self.ctx.store_meta.lock().unwrap();
+            meta.set_region(
+                &self.ctx.coprocessor_host,
+                sw.region,
+                &mut self.fsm.peer,
+                RegionChangeReason::SwitchWitness,
+            );
+        }
+        for s in sw.switches {
+            let (peer_id, is_witness) = (s.get_peer_id(), s.get_is_witness());
+            if self.fsm.peer_id() == peer_id {
+                if is_witness && !self.fsm.peer.is_leader() {
+                    let _ = self.fsm.peer.get_store().clear_data();
+                    self.fsm.peer.raft_group.set_priority(-1);
```

Comment:
when will it set back to normal?

Reply:
Should do that for snapshot as well, please check commentted code in `on_persist_snapshot`